src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -216,25 +217,49 @@ private void updateGridMode() {
     sendChannelFocus();
   }
 
+  // When turning a knob that sometimes controls Hue, sometimes Brightness,
+  // sometimes Saturation, look up the current mode of the surface and return
+  // the appropriate subparameter that this knob currently controls.
+  private CompoundParameter getColorSubparam(ColorParameter colorParameter) {
+    if (this.shiftOn) {
+      if (this.bankLeftOn || this.bankRightOn) {
```

Comment:
Feels sort of weird that in shift-mode, both bank buttons turn it to hue and nothing does brightness, but I guess I see no problem with that.

Reply:
It makes more sense if you think of it like, in either bank mode, holding shift momentarily disables the bank mode and makes the knob do its normal thing (hue), and for backwards compatibility with existing UX, holding shift when you're not in a bank mode momentarily makes the knob control saturation.